src/HydraAuction/Tx/Deposit.hs

```diff
@@ -0,0 +1,138 @@
+module HydraAuction.Tx.Deposit (
+  mkDeposit,
+  claimDeposit,
+) where
+
+-- Prelude imports
+import Prelude
+
+-- Haskell imports
+import Control.Monad (void)
+
+-- Plutus imports
+import Plutus.V2.Ledger.Api (fromData)
+
+-- Cardano node imports
+import Cardano.Api.UTxO qualified as UTxO
+
+-- Hydra imports
+import Hydra.Cardano.Api (
+  lovelaceToValue,
+  toPlutusData,
+  toPlutusKeyHash,
+  txOutDatum,
+  verificationKeyHash,
+  pattern ReferenceScriptNone,
+  pattern ShelleyAddressInEra,
+  pattern TxMintValueNone,
+  pattern TxOut,
+  pattern TxOutDatumInline,
+ )
+
+-- Hydra auction imports
+import HydraAuction.Addresses (VoucherCS (..))
+import HydraAuction.OnChain (AuctionScript (..), policy)
+import HydraAuction.Runner (Runner)
+import HydraAuction.Tx.Common (
+  actorTipUtxo,
+  addressAndKeys,
+  minLovelace,
+  mkInlineDatum,
+  mkInlinedDatumScriptWitness,
+  scriptAddress,
+  scriptPlutusScript,
+  scriptUtxos,
+ )
+import HydraAuction.Types (
+  AuctionTerms (..),
+  BidDepositDatum (..),
+  BidDepositRedeemer (..),
+ )
+import HydraAuctionUtils.Extras.Plutus (scriptCurrencySymbol)
+import HydraAuctionUtils.Monads (
+  logMsg,
+ )
+import HydraAuctionUtils.Tx.AutoCreateTx (
+  AutoCreateParams (..),
+  autoSubmitAndAwaitTx,
+ )
+import HydraAuctionUtils.Tx.Utxo (
+  filterAdaOnlyUtxo,
+ )
+
+mkDeposit :: AuctionTerms -> Runner ()
+mkDeposit terms = do
+  logMsg "Doing bidder deposit"
+
+  depositAddress <- scriptAddress Deposit terms
+
+  (bidderAddress, bidderVk, bidderSk) <- addressAndKeys
+
+  bidderMoneyUtxo <- filterAdaOnlyUtxo <$> actorTipUtxo
+
+  let mp = policy terms
+      voucerCS = VoucherCS $ scriptCurrencySymbol mp
+      bidDepositDatum = BidDepositDatum (toPlutusKeyHash $ verificationKeyHash bidderVk) voucerCS
+      bidDepositTxOut =
+        TxOut
+          (ShelleyAddressInEra depositAddress)
+          depositValue
+          (mkInlineDatum bidDepositDatum)
+          ReferenceScriptNone
+      -- FIXME: take as input?
```

Comment:
Should we use a fixed deposit value, or should we allow bidders to pick their own?

Reply:
They should pick.

Reply:
And the logic for selection of bidders will be completely off-chain.